octobot_tentacles_manager/base_worker/tentacle_worker.py

```diff
@@ -0,0 +1,199 @@
+        available_tentacle_data = await load_tentacle_with_metadata(self.tentacle_path)
+        tentacle_global_config = GlobalTentacleConfiguration()
+        await tentacle_global_config.read_config()
+        await tentacle_global_config.fill_tentacle_config(available_tentacle_data, self.default_tentacle_config)
+        await tentacle_global_config.save_config()
+
+    def log_summary(self):
+        if self.errors:
+            self.logger.info(" *** Error summary: ***")
+            for error in self.errors:
+                self.logger.error(f"Error when handling tentacle: {error}")
+        else:
+            self.logger.info(" *** All tentacles have been successfully processed ***")
+
+    def register_to_process_tentacles_modules(self, to_process_tentacle_data):
+        self.to_process_tentacle_modules = self._get_version_by_tentacle_data(to_process_tentacle_data)
+
+    async def handle_requirements(self, tentacle_data, callback):
+        missing_requirements = self._find_tentacles_missing_requirements(tentacle_data,
+                                                                         self.to_process_tentacle_modules)
+        if missing_requirements:
+            if not self.fetching_requirements:
+                self.fetching_requirements = True
+                await self._fetch_all_available_tentacles()
+            else:
+                await wait_for(self.requirements_downloading_event.wait(), self.TENTACLES_FETCHING_TIMEOUT)
+            await callback(tentacle_data, missing_requirements)
+
+    def confirm_action(self, action):
+        if not self.use_confirm_prompt:
+            return True
+        else:
+            confirmations = ["yes", "ye", "y", "oui", "o", "1"]
+            user_input = input(f"{action} Y/N").lower()
+            if user_input in confirmations:
+                return True
+            else:
+                self.logger.info("Action aborted.")
+                return False
+
+    async def _fetch_all_available_tentacles(self):
+        # try getting it from available tentacles
+        await gather(*[self._fetch_tentacles_for_requirement(repo)
+                       for repo in self._get_available_tentacles_repos()])
+        self.requirements_downloading_event.set()
+
+    async def _fetch_tentacles_for_requirement(self, repo):
+        await fetch_and_extract_tentacles(repo, DEFAULT_TENTACLES_URL, self.aiohttp_session, merge_dirs=True)
+        requirements_tentacles_path = path.join(TENTACLES_REQUIREMENTS_INSTALL_TEMP_DIR, TENTACLES_ARCHIVE_ROOT)
+        self.fetched_for_requirements_tentacles = await load_tentacle_with_metadata(requirements_tentacles_path)
+        self.fetched_for_requirements_tentacles_versions = \
+            self._get_version_by_tentacle_data(self.fetched_for_requirements_tentacles)
+
+    def _get_available_tentacles_repos(self):
+        # TODO: add advanced tentacles repos
+        return [TENTACLES_REQUIREMENTS_INSTALL_TEMP_DIR]
```

Comment:
We should put this config in an octobot global config I think.

Reply:
the path of the temporary folder containing the extracted tentacle modules ?